Vertexvis/vertex-web-sdk
# [PLAT-3317] Improve Scene documentation + general generated doc improvements

## Summary

- Adds documentation and examples around the following:
  - The `Scene.items()` method and the various queries and operations
  - The `Scene.raycaster()` method and `Raycaster`methods
  - The `Scene.camera()` method and `Camera` methods
  - The `Scene.crossSectioning()` method and `CrossSectioner` methods
- Fixes an issue with our `html` code examples and multiline comments in our components.
- Updates a few types to reflect that async functions can be used to fix failures in Sonar

## Test Plan

- Inspect the new comment output in the `components.d.ts` file and verify everything looks as expected
- Inspect the new documentation around the above methods as well as some multiline examples
  - This will have to be local until the next release, but running `yarn build; yarn generate:docs; cd docs; http-server` should spin up the docs
  - Example of an incorrect multiline comment that has been fixed: https://vertexvis.github.io/vertex-web-sdk/interfaces/Viewer.Components.VertexViewerWalkModeTool.html#teleportMode

## Release Notes

N/A

## Possible Regressions

Typedoc generated docs

## Dependencies

N/A
